src/pip_sync_faster/sync.py

```diff
@@ -0,0 +1,50 @@
+import hashlib
+import json
+import sys
+from os import environ
+from os.path import abspath
+from pathlib import Path
+from subprocess import run
+
+
+def get_hash(path):
+    """Return the hash of the given file."""
+    hashobj = hashlib.sha512()
```

Comment:
You could probably squeeze a few microseconds switching to sha256, if git considers that future-proof should be enough for this.

It doesn't probably matter.